spring-cloud-stream-metrics/src/main/java/org/springframework/cloud/stream/metrics/RelaxedPropertiesUtils.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.metrics;
+
+import org.springframework.boot.bind.RelaxedNames;
+
+/**
+ * @author Vinicius Carvalho
+ * Utiliy class to deal with {@link RelaxedNames}
+ */
+public class RelaxedPropertiesUtils {
+
+	/**
+	 * Searches relaxed names and try to find a canonical form (dot notation) of a property.
+	 *
+	 * For example, if a new list was built with JAVA_HOME as property, the return of this method would be java.home
+	 *
+	 * @param names - List of possible permutation of a property
+	 * @return The canonical form (dot notation) of this property
+	 */
+	public static String findCanonicalFormat(RelaxedNames names){
+		String canonicalForm = null;
+		for(String name : names){
+			if(name.contains(".") && upperCaseRatio(name) < 0.5){
```

Comment:
I see what you are saying. And yes it is totally arbitrary, the reason being cases such as OS, where Os would need to be catch. Initially I was lowering all entries that had ".", but we would endup with "*.contenttype.*" and not "*.contentType.*". 

In hindsight, it may be ok to just lowercase everything and not even bother about this pseudo heuristic, as the main reason for this is to have a canonical form downstream. So as long as every metric uses `contenttype` and not `Content-Type` or `contentType` we should be fine.